pkg/prometheus/server/operator.go

```diff
@@ -1199,6 +1203,24 @@ func (c *Operator) createOrUpdateWebConfigSecret(ctx context.Context, p *monitor
 	return nil
 }
 
+func (c *Operator) createOrUpdateThanosConfigSecret(ctx context.Context, p *monitoringv1.Prometheus) error {
+	prometheusHTTPClientConfig := newPrometheusHTTPClientConfig(p)
+
+	secret, err := prometheusHTTPClientConfig.createOrUpdatePrometheusHTTPClientConfigSecret()
+	if err != nil {
+		return fmt.Errorf("fialed to update Thanos HTTP client config secret: :%w", err)
+	}
+
+	operator.UpdateObject(
+		secret,
+		operator.WithLabels(c.config.Labels),
+		operator.WithAnnotations(c.config.Annotations),
+		operator.WithManagingOwner(p),
+	)
+
+	return k8sutil.CreateOrUpdateSecret(ctx, c.kclient.CoreV1().Secrets(p.Namespace), secret)
```

Comment:
(nit) it doesn't change anything but it's more coherent to use the secret's namespace.
```suggestion
	return k8sutil.CreateOrUpdateSecret(ctx, c.kclient.CoreV1().Secrets(secret.Namespace), secret)
```